src/renderer/pages/Onboarding/Vault/Vault.tsx

```diff
@@ -39,16 +68,33 @@ const Vault = ({ isOpen, onClose, onComplete }: Props) => {
     }
   }, [isOpen]);
 
+  useEffect(() => {
+    if (!qrPayload) return;
+
+    if (isDynamicDerivationSupport(qrPayload[0])) {
+      setQrType(QrCodeType.DYNAMIC_DERIVATIONS);
+
+      return;
+    }
+
+    if (
```

Comment:
Can we check `qrPayload.length > 1` and go to `setQrType(QrCodeType.MULTISHARD);`
Otherwise it's `setQrType(QrCodeType.SINGLESHARD);`?

Reply:
No, we should use such condition to support old multishard accounts